launch/launch/logging/__init__.py

```diff
@@ -331,7 +333,12 @@ def get_screen_handler():
     See launch_config() documentation for screen logging configuration.
     """
     if launch_config.screen_handler is None:
-        launch_config.screen_handler = handlers.StreamHandler(sys.stdout)
+        stream = codecs.StreamWriter(sys.stdout, errors='replace')
+        stream.encode = lambda msg, errors='replace': (
```

Comment:
@dirk-thomas hmm can't we use `codecs.getwriter(locale.getpreferredencoding(False))` to get the appropriate `StreamWriter` subclass instead of patching an instance?

Reply:
I am not sure how you can use that API to replace the existing logic. Can you please post the snippet you have in mind which sets the `launch_config.screen_handler`.

Reply:
Nevermind, this suggestion only applies if `stream.encode` actually does encode.